### DISCUSSION TITLE
Type 'string' is not assignable to type 'never' when using useFieldArray with TS

### DISCUSSION THREAD

**Discussion by imed-computas** [react-hook-form/react-hook-form#7260] [Q&A]
I am trying to use useFieldArray to keep track of to arrays in a component which is part of a bigger form with strong typing. I take control and register of the validate ts type in as parameteres, but when i try to specify name for my useFieldArrays i get the the error message saying "Type 'string' is not assignable to type 'never'  The expected type comes from property 'name' which is declared here on type 'UseFieldArrayProps<OurType, never, "id">'

Anyone experiencing that has experienced the same issue and found a solution?

**Comment by JamesWong1999** [comment]
can you share a simple codesandbox?

**Reply by rafaelfaria** [reply]
I am having the same problem.

```
  const {
    fields: urlFields,
    append: urlAppend,
    remove: urlRemove
  } = useFieldArray({ name: 'url', control });

```

**Reply by Moshyfawn** [reply]
> can you share a simple codesandbox?

Same applies, please, try to create an isolated [codessandbox](https://codesandbox.io/s/react-hook-form-v7-ts-template-4tn8i) with the issue reproduction. We're not wizards here, sadly ;)

**Reply by Valerika** [reply]
@rafaelfaria @imed-computas  @Moshyfawn I had the same problem. 
You need to add types of your array of fields
https://codesandbox.io/s/compassionate-engelbart-8q55jt?file=/src/App.tsx

![image](https://user-images.githubusercontent.com/15037937/154487445-b4d15b21-c747-4348-970d-87ee5ed7a659.png)
![image](https://user-images.githubusercontent.com/15037937/154487617-47bafb87-cf48-4687-b984-7a8e04fea052.png)

**Comment by nimeshmaharjan1** [comment]
You need to have object[] not just string[] that was the issue for me

instead of emails: ['', ''] i had to do emails: [{email: ''}]

**Reply by prince-bansal** [reply]
Thanks man, this saved my day.

**Reply by mrstevedev** [reply]
saved my day too
